src/Documentation/Grain-Versioning/Compatible-grains.md

```diff
@@ -0,0 +1,22 @@
+# Compatible grains
+
+In this document will be described the notion of "backward compatible" and "fully compatible".
+
+## Backward compatible
+
+A grain interface version Vn is said to be backward compatible with Vm if:
+
+  - The name of the interface didn't change (or the overridden typecode)
+  - All public methods present in the Vm version are in the Vn version. __It is important that
+    the signatures of the methods inherited from Vm are not modified__.
```

Comment:
a note about message serialization would be useful too, since if you use our magic serializer, just by modifying a single private field in a class part of the object graph you send to a grain can make the serialization to break